### PR TITLE
[Feature] FedEx

### PR DESCRIPTION
**Pull request by joneswong**
Adapted the open-sourced FedEx, a weight-sharing inspired fed hpo method, to our package.

### CODE REVIEW

File: federatedscope/autotune/fedex/client.py [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+        for hyper in hyperparams:
+            for k, v in hyper.items():
+                cmd_args.append(k)
+                cmd_args.append(v)
+
+        self._cfg.defrost()
+        self._cfg.merge_from_list(cmd_args)
+        self._cfg.freeze()
+
+        self.trainer.ctx.setup_vars()
```
Comment (joneswong) [inline federatedscope/autotune/fedex/client.py:28]: @DavdGao hi dawei, please help me validate whether this line can actually update the hyper-parameters of the client. thx!

Comment (DavdGao) [reply federatedscope/autotune/fedex/client.py:28]: Executing the code in line 24-26 will change the attribute `self.trainer.ctx.cfg` and calling function `setup_vars` will update the hyper-parameters and partial variables. However, the variables like `model`, `data`, `device` in `self.trainer.ctx` won't change as they are already created within `self.trainer.ctx`.

File: federatedscope/core/configs/cfg_training.py [hunk 2]
```diff
@@ -22,6 +22,7 @@ def extend_training_cfg(cfg):
     cfg.optimizer.type = 'SGD'
     cfg.optimizer.lr = 0.1
     cfg.optimizer.weight_decay = .0
+    cfg.optimizer.mmt = .0
```
Comment (rayrayraykk) [inline federatedscope/core/configs/cfg_training.py:25]: I'm not sure the abbreviation for momentum is general enough.

**Review by rayrayraykk — COMMENTED** [review]
Looks good! Please see the inline comments, thanks.

File: federatedscope/core/configs/cfg_training.py (hunk 2 above)
Comment (joneswong) [reply federatedscope/core/configs/cfg_training.py]: changed to "momentum" accordingly

**Review by DavdGao — COMMENTED** [review]
looks good to me

**Review by rayrayraykk — APPROVED** [review]
Looks good to me.

**Pull request merged**